pkg/utils/k8s.go

```diff
@@ -207,3 +208,60 @@ func LoadYAML(filename string, i interface{}) (interface{}, error) {
 
 	return genericTypeVal, nil
 }
+
+// filterByNamespace returns a ResourceList of Pods and ServiceMonitors filtered by namespaces
+func FilterByNamespace(namespaces []corev1.Namespace, resources corev1.ResourceList, rClient client.Client) (error, corev1.ResourceList) {
+	var err error
+	if len(namespaces) == 0 {
+		// if no namespaces are passed, return resources across all namespaces
+		listOpts := []client.ListOption{
+			client.InNamespace(""),
+		}
+		err, resources = getResources(listOpts, resources, rClient)
+
+	} else if len(namespaces) == 1 {
+		//if passed a single namespace, return resources across that namespace
+		listOpts := []client.ListOption{
+			client.InNamespace(namespaces[0].ObjectMeta.Name),
+		}
+		err, resources = getResources(listOpts, resources, rClient)
+
+	} else if len(namespaces) > 1 {
+		//if more than one namespaces is passed, loop through and add all resources to the ResourceList
+		for _, ns := range namespaces {
+			listOpts := []client.ListOption{
+				client.InNamespace(ns.ObjectMeta.Name),
+			}
+			err, resources = getResources(listOpts, resources, rClient)
+		}
+	} else {
+		err = errors.New("unexpected length of []namespaces")
+	}
+	return err, resources
+}
+
+// getResources() is a helper function for FilterByNamespace(), it returns a ResourceList in the requested namespaces
+// the namespaces are preset in listOpts
+func getResources(listOpts []client.ListOption, resources corev1.ResourceList, rClient client.Client) (error, corev1.ResourceList) {
```

Comment:
Let's have getResources take a runtime Object.
```suggestion
func getResources(obj runtime.Object, listOpts []client.ListOption, rClient client.Client) (error, corev1.ResourceList) {
```

Instead of creating a podList or serviceMonitorList you'll just pass in that runtime object (in this case a list), and it will go fetch the results for you. So for example:

```
obj := &PodList
FilterByNamespace(obj, namespaces, rClient)
```
We can drop the resource list. I like the idea but we want to filter on a resource across multiple namespaces and filters.